File: ASP Core - Backend/Dtos/AuthorDTO.cs
using System.ComponentModel.DataAnnotations;

namespace aspcorebackend.Dtos {
    public class AuthorDTO {
        public int AuthorId { get; set; }
        [MaxLength(150, ErrorMessage = "Author's firstname can't exceed 150 characters")]
        [Required(ErrorMessage = "Author's firstname is required")]
        public string Firstname { get; set; } = null!;
        [MaxLength(150, ErrorMessage = "Author's lastname can't exceed 150 characters")]
        [Required(ErrorMessage = "Author's lastname is required")]
        public string Lastname { get; set; } = null!;
        public DateTime AddedAt { get; set; }
        public List<BookDTO> Books { get; set; } = [];

        public AuthorDTO() {
            AddedAt = DateTime.Now;
        }
    }
}

File: Spring Boot - Backend/src/main/java/com/booknav/springbackend/dto/AuthorDTO.java
package com.booknav.springbackend.dto;

import com.booknav.springbackend.dto.Book.BookDTO;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
public class AuthorDTO {
    private Integer authorId;
    @NotBlank(message = "Author's firstname is required")
    @Size(max = 100, message = "Author's firstname can't exceed 100 characters")
    private String firstName;
    @NotBlank(message = "Author's lastname is required")
    @Size(max = 100, message = "Author's lastname can't exceed 100 characters.")
    private String lastName;
    private List<BookDTO> books;
    private LocalDateTime addedAt;
}
